src/connector/LeverageMacroBase.sol

```diff
@@ -0,0 +1,393 @@
+    function Cdps(bytes32) external view returns (ICdpManagerData.Cdp memory);
+}
+
+interface IBadgerConnector {
+    function handleOperation(
+        address token, 
+        uint256 amount, 
+        uint256 fee, 
+        ILeverageMacroBase.LeverageMacroOperation memory operation
+    ) external;
+}
+
+/// @title Base implementation of the LeverageMacro
+/// @notice Do not use this contract as a end users
+/// @dev You must extend this contract and override `owner()` to allow this to work:
+/// - As a Clone / Proxy (Not done, prob you'd read `owner` from calldata when using clones-with-immutable-args)
+/// - As a deployed copy (LeverageMacroReference)
+/// - Via delegate call (LeverageMacroDelegateTarget)
+/// @custom:known-issue Due to slippage some dust amounts for all intermediary tokens can be left, since there's no way to ask to sell all available
+
+abstract contract LeverageMacroBase is ILeverageMacroBase {
+    using SafeERC20 for IERC20;
+
+    IBorrowerOperations public immutable borrowerOperations;
+    IActivePool public immutable activePool;
+    ICdpCdps public immutable cdpManager;
+    IEBTCToken public immutable ebtcToken;
+    ISortedCdps public immutable sortedCdps;
+    ICollateralToken public immutable stETH;
+    bool internal immutable willSweep;
+
+    bytes32 constant FLASH_LOAN_SUCCESS = keccak256("ERC3156FlashBorrower.onFlashLoan");
+
+    function owner() public virtual returns (address) {
+        revert("Must be overridden");
+    }
+
+    function _assertOwner() internal {
+        // Reference will compare to variable,
+        require(owner() == msg.sender, "Must be owner");
+    }
+
+    // Leverage Macro should receive a request and set that data
+    // Then perform the request
+
+    constructor(
+        address _borrowerOperationsAddress,
+        address _activePool,
+        address _cdpManager,
+        address _ebtc,
+        address _coll,
+        address _sortedCdps,
+        bool _sweepToCaller
+    ) {
+        borrowerOperations = IBorrowerOperations(_borrowerOperationsAddress);
+        activePool = IActivePool(_activePool);
+        cdpManager = ICdpCdps(_cdpManager);
+        ebtcToken = IEBTCToken(_ebtc);
+        stETH = ICollateralToken(_coll);
+        sortedCdps = ISortedCdps(_sortedCdps);
+
+        willSweep = _sweepToCaller;
+    }
+
+    /**
+     * FL Setup
+     *         - Validate Caller
+     *
+     *         FL
+     *         - SwapsBefore
+     *         - Operation
+     *         - SwapsAfter
+     *         - Repay
+     *
+     *         - Post Operation Checks
+     *
+     *         - Sweep
+     */
+    /// @notice Entry point for the Macro
+    function doOperation(
+        FlashLoanType flType,
+        uint256 borrowAmount,
+        LeverageMacroOperation calldata operation,
+        PostOperationCheck postCheckType,
+        PostCheckParams calldata checkParams
+    ) external {
+        // Figure out the expected CDP ID using sortedCdps.toCdpId
```

Comment:
Missing `_assertOwner();`

This allows to perform arbitrary calls, so it's ultra dangerous

Am I missing something?